### PR TITLE
Add preset for diggy-danger-ores.

### PR DESCRIPTION
**Pull request by grilledham**

### CODE REVIEW

**Comment by Claude477** [comment]
no need with "require 'map_gen.maps.diggy.scenario'.register()" ?

**Comment by Claude477** [comment]
ore_pattern = require 'map_gen.maps.diggy.orepattern.tendrils_impure' need enabled normal.lua

**Comment by Claude477** [comment]
ore_pattern = require 'map_gen.maps.diggy.orepattern.clusters', need enabled in danger_ore.lua

File: map_gen/maps/diggy/presets/danger_ores.lua [hunk 1]
```diff
@@ -0,0 +1,482 @@
...
+            -- creates compound clusters of ores defined by a layered ore-gen
+            cluster_mode = true,
+            -- spawns tendrils of ore with roughly 80% purity
+            ore_pattern = require 'map_gen.maps.diggy.orepattern.tendrils_impure'
+
+            -- spawns some smaller dedicated and bigger mixed tendrils
+            --ore_pattern = require 'map_gen.maps.diggy.orepattern.tendrils',
+
+            -- spawns clusters of ore similar to vanilla, but mixed
+            --ore_pattern = require 'map_gen.maps.diggy.orepattern.clusters',
```
Comment (Claude477) [inline map_gen/maps/diggy/presets/danger_ores.lua]: this need check to enabled

**Comment by grilledham** [comment]
> no need with "require 'map_gen.maps.diggy.scenario'.register()" ?

Both the presets have .register()

> ore_pattern = require 'map_gen.maps.diggy.orepattern.tendrils_impure' need enabled normal.lua

normal.lua already has `ore_pattern = require 'map_gen.maps.diggy.orepattern.tendrils_impure'`?

> ore_pattern = require 'map_gen.maps.diggy.orepattern.clusters', need enabled in danger_ore.lua

I took the config from the currently running game. Is danger ore supposed to be using clusters instead of tendrils_impure?

File: map_gen/maps/diggy/presets/danger_ores.lua (hunk 1 above)
Comment (grilledham) [reply map_gen/maps/diggy/presets/danger_ores.lua]: You enable it by uncommenting it. If you want another preset with different config then make a copy and change it.

File: map_gen/maps/diggy/presets/danger_ores.lua [hunk 2]
```diff
@@ -0,0 +1,482 @@
...
+                ['crude-oil'] = 57
+            },
+            -- ==============================
+            -- Compound cluster mode settings
+            -- ==============================
+
+            -- creates compound clusters of ores defined by a layered ore-gen
+            cluster_mode = true,
+            -- spawns tendrils of ore with roughly 80% purity
+            ore_pattern = require 'map_gen.maps.diggy.orepattern.tendrils_impure'
```
Comment (Claude477) [inline map_gen/maps/diggy/presets/danger_ores.lua]: to comment this line[it used normal.lua]

File: map_gen/maps/diggy/presets/danger_ores.lua (hunk 1 above)
Comment (Claude477) [reply map_gen/maps/diggy/presets/danger_ores.lua]: yeah is need enabled [ore_pattern = require 'map_gen.maps.diggy.orepattern.clusters',]

**Pull request merged**